lib/core/include/irods/rodsError.h

```diff
@@ -33,7 +33,10 @@ typedef struct ErrorStack {
     struct ErrorMessage **errMsg;
 } rError_t;
 
-void allocateRErrorMsgIfNecessary(struct ErrorStack** myError);
+/// \brief Allocate memory for the ErrorStack struct if it is uninitialized
+///
+/// \param[in/out] myError the ErrorStack to be initialized
+void initialize_error_stack_if_necessary(struct ErrorStack** myError);
```

Comment:
Consider replacing `myError` with something else.

How about `_stack`?

This aligns with the function signature of `pop_error_message` at the bottom of this file.

Don't forget to update the documentation to match whatever you pick.